src/Sepes.Infrastructure/Service/Study/StudyServiceBase.cs

```diff
@@ -28,24 +29,37 @@ public StudyServiceBase(SepesDbContext db, IMapper mapper, ILogger logger, IUser
 
         public async Task<StudyDto> GetStudyDtoByIdAsync(int studyId, UserOperation userOperation)
         {
-            var studyFromDb = await GetStudyByIdAsync(studyId, userOperation, false);
+            var studyFromDb = await _studyModelService.GetByIdAsync(studyId, userOperation);
             var studyDto = _mapper.Map<StudyDto>(studyFromDb);
-
             return studyDto;
         }
 
         public async Task<StudyDetailsDto> GetStudyDetailsAsync(int studyId)
         {
-            var studyFromDb = await _studyModelService.GetStudyForStudyDetailsAsync(studyId);
+            var spElapsed = Stopwatch.StartNew();
+            var studyFromDb = await _studyModelService.GetForStudyDetailsAsync(studyId);
+            var elapsedDb = spElapsed.ElapsedMilliseconds;
+            spElapsed.Restart();
 
             var studyDetailsDto = _mapper.Map<StudyDetailsDto>(studyFromDb);
+            var elapsedDto = spElapsed.ElapsedMilliseconds;
+            spElapsed.Restart();
+
             await _studyLogoService.DecorateLogoUrlWithSAS(studyDetailsDto);
+            var elapsedUrl1 = spElapsed.ElapsedMilliseconds;
+            spElapsed.Restart();
+
+
             await StudyPermissionsUtil.DecorateDto(_userService, studyFromDb, studyDetailsDto.Permissions);
+            var elapsedUrl2 = spElapsed.ElapsedMilliseconds;
+            spElapsed.Restart();
 
             foreach (var curDs in studyDetailsDto.Datasets)
             {
                 curDs.Sandboxes = curDs.Sandboxes.Where(sd => sd.StudyId == studyId).ToList();
             }
+            var elapsedFiltered = spElapsed.ElapsedMilliseconds;
+            spElapsed.Restart();
 
```

Comment:
What are the purpose of these timers?

Reply:
Ah, those should not have made it out, removed it now